src/game/combat.h

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2020 uwadmin12
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <unordered_set>
+#include <map>
+
+
+#include "effect.h"
+#include "types.h"
+#include "object/creature.h"
+
+#include "SDL2/SDL_timer.h"
+
+namespace reone {
+
+namespace game {
+
+class Area;
+class Party;
+
+template <typename T>
+struct TimerPairCompare {
+    constexpr bool operator()(
+        std::pair<uint32_t, T> const& a,
+        std::pair<uint32_t, T> const& b)
+        const noexcept {
+        return a.first > b.first;
+    } // min heap!
+};
+
+template <typename T>
+struct TimerQueue {
```

Comment:
Given its generic nature, shouldn't this class be in libcommon?

Reply:
Good idea.